### ISSUE TITLE
Simplify the permutation issue

### ISSUE THREAD

**Issue by rreusser** [scijs/ndarray-lup-solve#1]
The decomposition is `LUx = Pb`. That means the solution is x = U^-1 L^-1 P b. You can solve this without permuting, but then the solution is permuted. I don't know a way to avoid this. I went with an optional work vector that is ugly but forward-compatible.

Is [ndarray-proxy](https://www.npmjs.com/package/ndarray-proxy) the correct solution here? Otherwise in-place permutations would be nice but are nontrivial.

**Comment by mikolalysenko** [comment]
ndarray-proxy is probably not the most performant way to do it.

I think just permuting the vector is the way to go.  We should have an ndarray-permute module for this I think.

**Comment by rreusser** [comment]
Agreed. I think I'll just make a placeholder module that does it the O(n) memory + O(n) operation way until we can do better. There might be a simplification here due to the cascading nature of the pivots, but since this would be roughly equivalent to argsort anyway, it'd certainly be nice to have.

**Comment by rreusser** [comment]
Hmm… no… it should be done with care à la ndarray-sort if it's gonna be done… probably just best to leave inline, ugly, and forward-compatible until it's done right.
